### PR TITLE
Improvements to quota doc.

### PR DESCRIPTION
**Pull request by erictune**
Address review comments in #11487

### CODE REVIEW

File: docs/admin/resource-quota.md [hunk 1]
```diff
@@ -42,8 +42,8 @@ work like this:
   support for making this mandatory via ACLs is planned.
 - Users put [compute resource limits](../user-guide/compute-resources.md) on their pods.
 - The administrator creates a Resource Quota for each namespace.
-- If users exceed the limits specified in the Resource Quota in a namespace, they are blocked
-  from creating more pods in that namespace.  The POST of the pod will fail with HTTP status
+- If creating a pod would cause the namespace to exceed any of the limits specified in the
+  the Resource Quota for that namespaces, then the request will fail with HTTP status
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: that namespaces -> that namespace

File: docs/admin/resource-quota.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ Examples of policies that could be created using namespaces and quotas are:
 - Limit the "testing" namespace to using 1 core and 1GiB RAM.  Let the "production" namespace
   use any amount.
 
-In the case where the total capacity of the cell is less than the sum of the quotas of the namespaces,
+In the case where the total capacity of the clustecluster less than the sum of the quotas of the namespaces,
```
Comment (davidopp) [inline docs/admin/resource-quota.md]: clustercluster -> cluster is

**Comment by erictune** [comment]
fixed.

**Comment by davidopp** [comment]
LGTM

**Comment by davidopp** [comment]
NOTE: MERGE THIS **AFTER** #11487

**Pull request merged**